server/src/test/java/org/elasticsearch/cluster/metadata/TemplateUpgradeServiceTests.java

```diff
@@ -75,8 +81,25 @@
 
 public class TemplateUpgradeServiceTests extends ESTestCase {
 
-    private final ClusterService clusterService = new ClusterService(Settings.EMPTY, new ClusterSettings(Settings.EMPTY,
-        ClusterSettings.BUILT_IN_CLUSTER_SETTINGS), null, Collections.emptyMap());
+    private ThreadPool threadPool;
+    private ClusterService clusterService;
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
```

Comment:
this is a personal preference, I like to avoid overriding the setup and teardown methods of estestcase and use separate one

Reply:
Makes perfect sense, calling the super implementation is cumbersome and forgetting is dangerous.